packages/core/src/components/Autocomplete/Autocomplete.e2e.test.js

```diff
@@ -1,19 +1,113 @@
-import { getElementByClassName } from '../../helpers/e2e';
 
 const rootURL = `${ROOT_URL}/example/components.autocomplete.react/`;
 
-describe('Alert component', () => {
+describe('Autocomplete component', () => {
   it('Should render', async() => {
     await driver.get(rootURL);
 
     const el = await getElementByClassName('ds-u-clearfix ds-c-autocomplete');
     expect(el).toBeTruthy();
   });
 
+  it('Should expand the listbox when keys are pressed', async() => {
+    await driver.get(rootURL);
+
+    let el = await getElementByXPath('//*[@id="autocomplete_1"]');
+    el.click();
+    await el.sendKeys('c');
+
+    el = await getElementByXPath('//*[@id="autocomplete_owned_container_4"]');
+    expect(el).toBeTruthy();
+  });
+
+  it('Should set the input value correctly when a listbox selection is clicked', async() => {
+    await driver.get(rootURL);
+
+    let el = await getElementByXPath('//*[@id="autocomplete_1"]');
+    el.click();
+    await el.sendKeys('c');
+
+    el = await getElementByXPath('//*[@id="downshift-0-item-0"]');
+    el.click();
+
+    el = await getElementByXPath('//*[@id="autocomplete_1"]');
+    el = await el.getAttribute('value');
+    expect(el).toEqual('Cook County, IL');
+  });
+
+  it('Should set the input value to empty when Clear search is clicked', async() => {
+    await driver.get(rootURL);
+
+    let el = await getElementByXPath('//*[@id="autocomplete_1"]');
+    el.click();
+    await el.sendKeys('c');
+
+    el = await getElementByXPath('//*[@id="downshift-0-item-0"]');
+    el.click();
+
+    el = await getElementByXPath('//*[@id="js-example"]/div/div[1]/button');
+    el.click();
+
+    el = await getElementByXPath('//*[@id="autocomplete_1"]');
+    el = await el.getAttribute('value');
+    expect(el).toEqual('');
+  });
+
+  it('Should select list items by keyboard', async() => {
+    await driver.get(rootURL);
+
+    let el = await getElementByXPath('//*[@id="autocomplete_1"]');
+    el.click();
+    await el.sendKeys('c');
+    await el.sendKeys(key.ARROW_DOWN);
+    await el.sendKeys(key.ENTER);
+
+    el = await el.getAttribute('value');
+    expect(el).toEqual('Cook County, IL');
+  });
+
+  it('Should clear the input value by keyboard', async() => {
+    await driver.get(rootURL);
+
+    let el = await getElementByXPath('//*[@id="autocomplete_1"]');
+    el.click();
+    await el.sendKeys('c');
+    await el.sendKeys(key.ARROW_DOWN);
+    await el.sendKeys(key.ENTER);
+    await el.sendKeys(key.TAB);
+
+    el = await getElementByXPath('//*[@id="js-example"]/div/div[1]/button');
+    el.click();
+
+    el = await getElementByXPath('//*[@id="autocomplete_1"]');
+    el = await el.getAttribute('value');
+    expect(el).toEqual('');
+  });
+
+  it('Closes the listbox when ESC is pressed', async() => {
+    await driver.get(rootURL);
+
+    let container = await getElementByXPath(
+      '//*[@id="js-example"]/div/div[1]/div'
+    );
+    container = await container.getAttribute('outerHTML');
```

Comment:
To your question about using the outerHTML to test the presence of the element, I think it would be better to use `findElements`. Something like

```js
const autocompleteField = await getElementByXPath('//*[@id="autocomplete_1"]');

// Make sure the list shows when we start typing
autocompleteField.click();
await autocompleteField.sendKeys('c');
let listbox = await driver.findElements(by.css('the-listbox-selector'))
expect(listbox.length).toEqual(1);

// Make sure it disappears when we hit escape
await autocompleteField.sendKeys(key.ESCAPE);
listbox = await driver.findElements(by.css('the-listbox-selector'))
expect(listbox.length).toEqual(0);
```

Reply:
I also think it would make the tests easier to follow if you use the variable name to describe what the element is rather than reusing the `el` variable.

Reply:
@pwolfert I took your advice and named the test `elem` more specifically. They read a lot better now.

I also found a way to assert keyboard focus effectively. It could use for a refactor to a helper function, but I'll do that as an enhancement since `Autocomplete` is the only one doing this style of testing atm.

Reply:
They do read a lot easier 👍 LGTM